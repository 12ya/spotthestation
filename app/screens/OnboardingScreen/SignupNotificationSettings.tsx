import React from "react"
import { PressableProps, TextStyle, View, ViewStyle } from "react-native"
import { Text, Button, Icon, Toggle } from "../../components"
import { colors, fontSizes, lineHeights, scale, typography } from "../../theme"

export interface SignupNotificationSettingsProps {
  /**
   * Is notifications on/off
   */
  value: boolean
  /**
   * Action for a change notification settings
   */
  onValueChange: (value: boolean) => void
  /**
   * Action to move to the next step.
   */
  onAction: PressableProps["onPress"]
}

export function SignupNotificationSettings({
  value,
  onValueChange,
  onAction,
}: SignupNotificationSettingsProps) {
  return (
    <>
      <Text
        accessible
        accessibilityLabel="detecting"
        accessibilityHint="detecting current location"
        accessibilityRole="text"
        tx="onboarding.completeProfile.notification.tittle"
        style={$title}
      />
      <View style={$valueContainer}>
        <Icon icon="tv" size={24} containerStyle={$icon} />
        <View style={$switchContainer}>
          <View
            accessible
            accessibilityLabel="notification"
            accessibilityHint="notification"
            accessibilityRole="text"
            style={$labelContainer}
          >
            <Text tx="onboarding.completeProfile.notification.label" style={$label} />
            <Text tx="onboarding.completeProfile.notification.tip" style={$tip} />
          </View>
          <Toggle
            accessible
            accessibilityLabel="switch button"
            accessibilityHint="toggle notifications"
            variant="switch"
            value={value}
            onValueChange={onValueChange}
          />
        </View>
      </View>
      <Button
        accessible
        accessibilityLabel="next button"
        accessibilityHint="navigate to location detection"
        tx="onboarding.completeProfile.notification.nextButton"
        pressedStyle={$button}
        style={$button}
        textStyle={$buttonText}
        onPress={onAction}
      />
    </>
  )
}

const $title: TextStyle = {
  color: colors.palette.neutral250,
  fontSize: fontSizes[36],
  fontFamily: typography.primary.normal,
  lineHeight: lineHeights[42],
  paddingBottom: scale(36),
}

const $button: ViewStyle = {
  width: "100%",
  height: scale(56),
  backgroundColor: colors.palette.buttonBlue,
  borderRadius: scale(28),
  borderWidth: 0,
}

const $buttonText: TextStyle = {
  color: colors.palette.neutral100,
  fontSize: fontSizes[18],
  fontFamily: typography.primary.medium,
  lineHeight: lineHeights[21],
}

const $valueContainer: ViewStyle = {
  flexDirection: "row",
  marginBottom: scale(36),
}

const $switchContainer: ViewStyle = {
  flexDirection: "row",
  justifyContent: "space-between",
  paddingBottom: scale(18),
  borderBottomWidth: scale(1),
  borderBottomColor: colors.palette.neutral350,
  flex: 1,
}

const $labelContainer: ViewStyle = {
  width: "70%",
}

const $label: TextStyle = {
  color: colors.palette.neutral250,
  fontSize: fontSizes[16],
  fontFamily: typography.primary.normal,
  lineHeight: lineHeights[21],
}

const $tip: TextStyle = {
  color: colors.palette.neutral450,
  fontSize: fontSizes[16],
  fontFamily: typography.primary.light,
  lineHeight: lineHeights[21],
}

const $icon: ViewStyle = {
  marginRight: scale(18),
}
