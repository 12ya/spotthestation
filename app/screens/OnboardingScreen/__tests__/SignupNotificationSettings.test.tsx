/* eslint-disable react/display-name */
/* eslint-disable @typescript-eslint/no-unsafe-call */
/* eslint-disable @typescript-eslint/no-unsafe-member-access */
import { NavigationContainer } from "@react-navigation/native"
import React from "react"
import { SignupNotificationSettings } from "../SignupNotificationSettings"
import { render } from "@testing-library/react-native"

it("renders correctly", () => {
  const tree = render(
    <NavigationContainer>
      <SignupNotificationSettings value={true} onAction={() => ({})} onValueChange={() => ({})} />
    </NavigationContainer>,
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
