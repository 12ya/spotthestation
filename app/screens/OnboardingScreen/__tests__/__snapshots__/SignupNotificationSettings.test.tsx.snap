// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
[
  <Text
    accessibilityHint="detecting current location"
    accessibilityLabel="detecting"
    accessibilityRole="text"
    accessible={true}
    allowFontScaling={false}
    style={
      [
        {},
        [
          {
            "fontSize": 23,
            "lineHeight": 35,
          },
          {
            "fontFamily": "Helvetica Neue",
          },
          {
            "color": "#191015",
          },
        ],
        undefined,
        undefined,
        {
          "color": "#DEDDDE",
          "fontFamily": "Helvetica Neue",
          "fontSize": 52,
          "lineHeight": 60,
          "paddingBottom": 52,
        },
      ]
    }
  >
    onboarding.completeProfile.notification.title undefined
  </Text>,
  <View
    style={
      {
        "flexDirection": "row",
        "marginBottom": 52,
      }
    }
  >
    <View
      style={
        {
          "marginRight": 26,
        }
      }
    >
      <RCTImageView
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        internal_analyticTag={null}
        resizeMode="contain"
        source={
          [
            {
              "height": undefined,
              "uri": undefined,
              "width": undefined,
            },
          ]
        }
        style={
          {
            "height": 35,
            "overflow": "hidden",
            "resizeMode": "contain",
            "width": 35,
          }
        }
      />
    </View>
    <View
      style={
        {
          "borderBottomColor": "#1E2022",
          "borderBottomWidth": 2,
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 26,
        }
      }
    >
      <View
        accessibilityHint="notification"
        accessibilityLabel="notification"
        accessibilityRole="text"
        accessible={true}
        style={
          {
            "width": "70%",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            [
              {},
              [
                {
                  "fontSize": 23,
                  "lineHeight": 35,
                },
                {
                  "fontFamily": "Helvetica Neue",
                },
                {
                  "color": "#191015",
                },
              ],
              undefined,
              undefined,
              {
                "color": "#DEDDDE",
                "fontFamily": "Helvetica Neue",
                "fontSize": 23,
                "lineHeight": 30,
              },
            ]
          }
        >
          onboarding.completeProfile.notification.label undefined
        </Text>
        <Text
          allowFontScaling={false}
          style={
            [
              {},
              [
                {
                  "fontSize": 23,
                  "lineHeight": 35,
                },
                {
                  "fontFamily": "Helvetica Neue",
                },
                {
                  "color": "#191015",
                },
              ],
              undefined,
              undefined,
              {
                "color": "#ADADAE",
                "fontFamily": "HelveticaNeue-Light",
                "fontSize": 23,
                "lineHeight": 30,
              },
            ]
          }
        >
          onboarding.completeProfile.notification.tip undefined
        </Text>
      </View>
      <View
        accessibilityHint="toggle notifications"
        accessibilityLabel="switch button"
        accessibilityRole="switch"
        accessibilityState={
          {
            "busy": undefined,
            "checked": true,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "borderRadius": 23,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "flexGrow": 0,
                  "flexShrink": 0,
                  "height": 40,
                  "justifyContent": "space-between",
                  "overflow": "hidden",
                  "width": 66,
                },
                {
                  "backgroundColor": "#353739",
                },
                undefined,
              ]
            }
          >
            <View
              animatedStyle={
                {
                  "value": {
                    "opacity": 1,
                  },
                }
              }
              collapsable={false}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#0B3D91",
                  "borderColor": "rgba(0, 0, 0, 0)",
                  "height": "100%",
                  "opacity": 1,
                  "overflow": "hidden",
                  "paddingEnd": 6,
                  "paddingStart": 6,
                  "position": "absolute",
                  "width": "100%",
                }
              }
            />
            <View
              animatedStyle={
                {
                  "value": {
                    "marginStart": -35,
                    "start": "100%",
                  },
                }
              }
              collapsable={false}
              style={
                {
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 17,
                  "height": 29,
                  "marginStart": -35,
                  "position": "absolute",
                  "start": "100%",
                  "width": 29,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>,
  <View
    accessibilityHint="navigate to location detection"
    accessibilityLabel="next button"
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        [
          {
            "alignItems": "center",
            "borderRadius": 6,
            "flexDirection": "row",
            "justifyContent": "center",
            "minHeight": 80,
            "overflow": "hidden",
            "paddingHorizontal": 17,
            "paddingVertical": 17,
          },
          {
            "backgroundColor": "#FFFFFF",
            "borderColor": "#B6ACA6",
            "borderWidth": 1,
          },
        ],
        {
          "backgroundColor": "#0B3D91",
          "borderRadius": 40,
          "borderWidth": 0,
          "height": 80,
          "width": "100%",
        },
        false,
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        [
          {},
          [
            {
              "fontSize": 23,
              "lineHeight": 35,
            },
            {
              "fontFamily": "Helvetica Neue",
            },
            {
              "color": "#191015",
            },
          ],
          undefined,
          undefined,
          [
            {
              "flexGrow": 0,
              "flexShrink": 1,
              "fontFamily": "HelveticaNeue-Medium",
              "fontSize": 23,
              "lineHeight": 29,
              "textAlign": "center",
              "zIndex": 2,
            },
            {
              "color": "#FFFFFF",
              "fontFamily": "HelveticaNeue-Medium",
              "fontSize": 26,
              "lineHeight": 30,
            },
            false,
          ],
        ]
      }
    >
      onboarding.completeProfile.notification.nextButton undefined
    </Text>
  </View>,
]
`;
